test/monitoring/aws-ec2/EC2Monitoring.test.ts

```diff
@@ -2,70 +2,90 @@ import { Stack } from "aws-cdk-lib";
 import { Template } from "aws-cdk-lib/assertions";
 import { AutoScalingGroup } from "aws-cdk-lib/aws-autoscaling";
 
-import { EC2Monitoring } from "../../../lib";
+import { EC2Monitoring, InstanceFamilyTypes } from "../../../lib";
 import { addMonitoringDashboardsToStack } from "../../utils/SnapshotUtil";
 import { TestMonitoringScope } from "../TestMonitoringScope";
 
-test("snapshot test: all instances, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: ASG, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    autoScalingGroup: AutoScalingGroup.fromAutoScalingGroupName(
-      stack,
-      "DummyASG",
-      "DummyASG"
-    ),
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: instance filter, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    instanceIds: ["instance1", "instance2"],
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: instance filter + ASG, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    instanceIds: ["instance1", "instance2"],
-    autoScalingGroup: AutoScalingGroup.fromAutoScalingGroupName(
-      stack,
-      "DummyASG",
-      "DummyASG"
-    ),
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
+function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O): K[] {
+  return Object.keys(obj).filter((k) => Number.isNaN(+k)) as K[];
```

Comment:
The filtering seems redundant.

Reply:
Not really because of how enums in Typescript work. I got this function from here: https://www.petermorlion.com/iterating-a-typescript-enum/.

I also found the [ts-enum-util](https://www.npmjs.com/package/ts-enum-util) library that handles this, which could be an alternative to hand-crafting my own function.

Another alternative would be to list the instance family types manually.

Reply:
Ah you're right, I was thinking of string enums.